src/org/jitsi/impl/neomedia/MediaStreamStatsImpl.java

```diff
@@ -1548,4 +1548,51 @@ public void addRembListener(REMBListener listener)
             }
         }
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * This method is different from {@link #getUploadRateKiloBitPerSec()} in
+     * that:
+     * 1. It is not necessary for {@link #updateStats()} to be called
+     * periodically by the user of libjitsi in order for it to return correct
+     * values.
+     * 2. The returned value is based on the average bitrate over a fixed
+     * window, as opposed to an EWMA.
+     * 3. The measurement is performed after the {@link MediaStream}'s
+     * transformations, notably after simulcast layers are dropped (i.e. closer
+     * to the network interface).
+     *
+     * The return value includes RTP payload and RTP headers, as well as RTCP.
+     */
+    @Override
+    public long getSendingBitrate()
+    {
+        long sbr = -1;
+        AbstractRTPConnector rtpConnector = mediaStreamImpl.getRTPConnector();
+
+        if (rtpConnector != null)
+        {
+            try
+            {
+                RTPConnectorOutputStream rtpStream
+                        = rtpConnector.getDataOutputStream(false);
+                RTPConnectorOutputStream rtcpStream
+                        = rtpConnector.getControlOutputStream(false);
+
+                if (rtpStream != null && rtcpStream != null)
```

Comment:
I didn't get it why both RTP and RTCP streams are a must. I guess it could be a requirement of RateStatistics but didn't really search for the cause of the requirement.